gRPC client doesn't send protocols for ALPN

## Environment Details
* Helidon Version: 4.2.4, 4.3.0-SNAPSHOT
* Helidon SE or Helidon MP SE
* JDK version: 21
* OS: Mac

----------

## Problem Description
When the gRPC Client is used to connect via TLS to a server, it is supposed to use ALPN to negotiate a protocol. However, ALPN negotiation to HTTP/2 (`h2`) fails, as the client does not advertise any ALPN protocols. See https://github.com/helidon-io/helidon/blob/2d4bf6b4f55a8f87e7b7ca51adc4b713284ce60a/webclient/grpc/src/main/java/io/helidon/webclient/grpc/GrpcBaseClientCall.java#L286-L291 This causes the connection to fail with standards-compliant gRPC servers such as the Netty gRPC server.

The problem is reproducible 100% of the time. The following output was collected from the client with `-Djavax.net.debug=ssl:handshake`. Notice the output:
```
javax.net.ssl|INFO|10|main|2025-07-27 20:51:15.935 EDT|AlpnExtension.java:179|No available application protocols
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.935 EDT|SSLExtensions.java:272|Ignore, context unavailable extension: application_layer_protocol_negotiation
```

```
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.666 EDT|SSLCipher.java:432|jdk.tls.keyLimits:  entry = AES/GCM/NoPadding KeyUpdate 2^37. AES/GCM/NOPADDING:KEYUPDATE = 137438953472
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.668 EDT|SSLCipher.java:432|jdk.tls.keyLimits:  entry =  ChaCha20-Poly1305 KeyUpdate 2^37. CHACHA20-POLY1305:KEYUPDATE = 137438953472
javax.net.ssl|WARNING|10|main|2025-07-27 20:51:15.934 EDT|ServerNameExtension.java:265|Unable to indicate server name
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.934 EDT|SSLExtensions.java:272|Ignore, context unavailable extension: server_name
javax.net.ssl|INFO|10|main|2025-07-27 20:51:15.935 EDT|AlpnExtension.java:179|No available application protocols
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.935 EDT|SSLExtensions.java:272|Ignore, context unavailable extension: application_layer_protocol_negotiation
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.935 EDT|SessionTicketExtension.java:352|Stateless resumption supported
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.936 EDT|SSLExtensions.java:272|Ignore, context unavailable extension: cookie
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.969 EDT|SSLExtensions.java:272|Ignore, context unavailable extension: renegotiation_info
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.969 EDT|PreSharedKeyExtension.java:659|No session to resume.
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.969 EDT|SSLExtensions.java:272|Ignore, context unavailable extension: pre_shared_key
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.971 EDT|ClientHello.java:638|Produced ClientHello handshake message (
"ClientHello": {
  "client version"      : "TLSv1.2",
  "random"              : "003F5A289556992B08234012E052AF3AED24A45B1BD79C91E9485CA023B6E5B0",
  "session id"          : "2F5F5AA9AB5B7926536F42709B1695EF73D9752D8885CDE275BE9734715DD4C8",
  "cipher suites"       : "[TLS_AES_256_GCM_SHA384(0x1302), TLS_AES_128_GCM_SHA256(0x1301), TLS_CHACHA20_POLY1305_SHA256(0x1303), TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384(0xC02C), TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256(0xC02B), TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256(0xCCA9), TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384(0xC030), TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256(0xCCA8), TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256(0xC02F), TLS_DHE_RSA_WITH_AES_256_GCM_SHA384(0x009F), TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256(0xCCAA), TLS_DHE_DSS_WITH_AES_256_GCM_SHA384(0x00A3), TLS_DHE_RSA_WITH_AES_128_GCM_SHA256(0x009E), TLS_DHE_DSS_WITH_AES_128_GCM_SHA256(0x00A2), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384(0xC024), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384(0xC028), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256(0xC023), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256(0xC027), TLS_DHE_RSA_WITH_AES_256_CBC_SHA256(0x006B), TLS_DHE_DSS_WITH_AES_256_CBC_SHA256(0x006A), TLS_DHE_RSA_WITH_AES_128_CBC_SHA256(0x0067), TLS_DHE_DSS_WITH_AES_128_CBC_SHA256(0x0040), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA(0xC00A), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA(0xC014), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA(0xC009), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA(0xC013), TLS_DHE_RSA_WITH_AES_256_CBC_SHA(0x0039), TLS_DHE_DSS_WITH_AES_256_CBC_SHA(0x0038), TLS_DHE_RSA_WITH_AES_128_CBC_SHA(0x0033), TLS_DHE_DSS_WITH_AES_128_CBC_SHA(0x0032), TLS_RSA_WITH_AES_256_GCM_SHA384(0x009D), TLS_RSA_WITH_AES_128_GCM_SHA256(0x009C), TLS_RSA_WITH_AES_256_CBC_SHA256(0x003D), TLS_RSA_WITH_AES_128_CBC_SHA256(0x003C), TLS_RSA_WITH_AES_256_CBC_SHA(0x0035), TLS_RSA_WITH_AES_128_CBC_SHA(0x002F), TLS_EMPTY_RENEGOTIATION_INFO_SCSV(0x00FF)]",
  "compression methods" : "00",
  "extensions"          : [
    "status_request (5)": {
      "certificate status type": ocsp
      "OCSP status request": {
        "responder_id": <empty>
        "request extensions": {
          <empty>
        }
      }
    },
    "supported_groups (10)": {
      "named groups": [x25519, secp256r1, secp384r1, secp521r1, x448, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192]
    },
    "ec_point_formats (11)": {
      "formats": [uncompressed]
    },
    "status_request_v2 (17)": {
      "cert status request": {
        "certificate status type": ocsp_multi
        "OCSP status request": {
          "responder_id": <empty>
          "request extensions": {
            <empty>
          }
        }
      }
    },
    "extended_master_secret (23)": {
      <empty>
    },
    "session_ticket (35)": {
      <empty>
    },
    "signature_algorithms (13)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, ed25519, ed448, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1]
    },
    "supported_versions (43)": {
      "versions": [TLSv1.3, TLSv1.2]
    },
    "psk_key_exchange_modes (45)": {
      "ke_modes": [psk_dhe_ke]
    },
    "signature_algorithms_cert (50)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, ed25519, ed448, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1]
    },
    "key_share (51)": {
      "client_shares": [  
        {
          "named group": x25519
          "key_exchange": {
            0000: 6F 0C EF 9B EE A4 19 7A   FD 0D 71 AB 52 B2 94 1B  o......z..q.R...
            0010: 1B 9A 9D DD D4 D6 E7 68   CA 5D A7 FD FA 94 8A 69  .......h.].....i
          }
        },
        {
          "named group": secp256r1
          "key_exchange": {
            0000: 04 5D 7F F1 7E 91 D5 D8   EA 1E 7E FD 2E 53 E1 E4  .]...........S..
            0010: 6C 4B 41 57 DB 1B F2 97   9D D5 30 D3 C0 1C F9 2A  lKAW......0....*
            0020: 2B 37 2E 06 74 ED 07 5F   0E 5E 6E C3 D8 B9 8F 99  +7..t.._.^n.....
            0030: 98 6D 7E DA AE 33 70 86   D8 3A E1 DF DF 55 F7 F9  .m...3p..:...U..
            0040: 56 
          }
        },
      ]
    }
  ]
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|ServerHello.java:892|Consuming ServerHello handshake message (
"ServerHello": {
  "server version"      : "TLSv1.2",
  "random"              : "7EEE6A6A93AA89FD80D0ABF878CB8E40A75A44C74ABD2B4A551520EF4EA956B2",
  "session id"          : "2F5F5AA9AB5B7926536F42709B1695EF73D9752D8885CDE275BE9734715DD4C8",
  "cipher suite"        : "TLS_AES_128_GCM_SHA256(0x1301)",
  "compression methods" : "00",
  "extensions"          : [
    "supported_versions (43)": {
      "selected version": [TLSv1.3]
    },
    "key_share (51)": {
      "server_share": {
        "named group": x25519
        "key_exchange": {
          0000: 92 47 3A EE 66 4E 3F 98   71 19 72 F5 73 E7 D3 24  .G:.fN?.q.r.s..$
          0010: 58 43 8B D8 8E F6 03 B3   24 39 3F AD 5B 8E AF 64  XC......$9?.[..d
        }
      },
    }
  ]
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|SSLExtensions.java:204|Consumed extension: supported_versions
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|ServerHello.java:988|Negotiated protocol version: TLSv1.3
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|SSLExtensions.java:175|Ignore unsupported extension: server_name
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|SSLExtensions.java:175|Ignore unsupported extension: max_fragment_length
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|SSLExtensions.java:175|Ignore unsupported extension: status_request
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|SSLExtensions.java:175|Ignore unsupported extension: ec_point_formats
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.982 EDT|SSLExtensions.java:175|Ignore unsupported extension: application_layer_protocol_negotiation
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.983 EDT|SSLExtensions.java:175|Ignore unsupported extension: status_request_v2
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.983 EDT|SSLExtensions.java:175|Ignore unsupported extension: extended_master_secret
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.983 EDT|SSLExtensions.java:175|Ignore unsupported extension: session_ticket
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.983 EDT|SSLExtensions.java:204|Consumed extension: supported_versions
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.983 EDT|SSLExtensions.java:204|Consumed extension: key_share
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.983 EDT|SSLExtensions.java:175|Ignore unsupported extension: renegotiation_info
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.983 EDT|PreSharedKeyExtension.java:922|Handling pre_shared_key absence.
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: server_name
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: max_fragment_length
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: status_request
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: ec_point_formats
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: application_layer_protocol_negotiation
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: status_request_v2
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: extended_master_secret
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: session_ticket
javax.net.ssl|WARNING|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:227|Ignore impact of unsupported extension: supported_versions
javax.net.ssl|WARNING|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:227|Ignore impact of unsupported extension: key_share
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: renegotiation_info
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.984 EDT|SSLExtensions.java:219|Ignore unavailable extension: pre_shared_key
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.986 EDT|SSLCipher.java:1836|KeyLimit read side: algorithm = AES/GCM/NoPadding:KEYUPDATE
countdown value = 137438953472
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.986 EDT|SSLCipher.java:1987|KeyLimit write side: algorithm = AES/GCM/NoPadding:KEYUPDATE
countdown value = 137438953472
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.987 EDT|ChangeCipherSpec.java:244|Consuming ChangeCipherSpec message
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.988 EDT|EncryptedExtensions.java:172|Consuming EncryptedExtensions handshake message (
"EncryptedExtensions": [
  "supported_groups (10)": {
    "named groups": [x25519, secp256r1, secp384r1, secp521r1, x448, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192]
  }
]
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.988 EDT|SSLExtensions.java:185|Ignore unavailable extension: server_name
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.988 EDT|SSLExtensions.java:185|Ignore unavailable extension: max_fragment_length
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.988 EDT|SSLExtensions.java:204|Consumed extension: supported_groups
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.988 EDT|SSLExtensions.java:219|Ignore unavailable extension: server_name
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.988 EDT|SSLExtensions.java:219|Ignore unavailable extension: max_fragment_length
javax.net.ssl|WARNING|10|main|2025-07-27 20:51:15.988 EDT|SSLExtensions.java:227|Ignore impact of unsupported extension: supported_groups
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.988 EDT|SSLExtensions.java:219|Ignore unavailable extension: application_layer_protocol_negotiation
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.989 EDT|CertificateRequest.java:974|Consuming CertificateRequest handshake message (
"CertificateRequest": {
  "certificate_request_context": "",
  "extensions": [
    "signature_algorithms (13)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, ed25519, ed448, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, ecdsa_sha1, rsa_pkcs1_sha1]
    },
    "signature_algorithms_cert (50)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, ed25519, ed448, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, ecdsa_sha1, rsa_pkcs1_sha1]
    }
  ]
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.989 EDT|SSLExtensions.java:204|Consumed extension: signature_algorithms
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.989 EDT|SSLExtensions.java:185|Ignore unavailable extension: certificate_authorities
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.989 EDT|SSLExtensions.java:204|Consumed extension: signature_algorithms_cert
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.989 EDT|SSLExtensions.java:236|Populated with extension: signature_algorithms
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.989 EDT|SSLExtensions.java:219|Ignore unavailable extension: certificate_authorities
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.989 EDT|SSLExtensions.java:236|Populated with extension: signature_algorithms_cert
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.990 EDT|CertificateMessage.java:1135|Consuming server Certificate handshake message (
"Certificate": {
  "certificate_request_context": "",
  "certificate_list": [  
  {
    "certificate" : {
      "version"            : "v3",
      "serial number"      : "0f:6d:18:7d:02:26:ca:9c",
      "signature algorithm": "SHA384withECDSA",
      "issuer"             : "O=PKWARE",
      "not before"         : "2024-05-22 13:04:25.000 EDT",
      "not  after"         : "2025-05-22 13:04:25.000 EDT",
      "subject"            : "O=PKWARE",
      "subject public key" : "EC",
      "extensions"         : [
        {
          ObjectId: 2.5.29.17 Criticality=false
          SubjectAlternativeName [
            IPAddress: 127.0.0.1
            DNSName: localhost
          ]
        },
        {
          ObjectId: 2.5.29.14 Criticality=false
          SubjectKeyIdentifier [
          KeyIdentifier [
          0000: 57 59 75 A1 7A 2D 27 B4   16 53 EF 3E 89 61 BC 10  WYu.z-'..S.>.a..
          0010: C2 D1 89 27                                        ...'
          ]
          ]
        }
      ]}
    "extensions": {
      <no extension>
    }
  },
]
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:15.990 EDT|SSLExtensions.java:185|Ignore unavailable extension: status_request
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:16.023 EDT|CertificateVerify.java:1169|Consuming CertificateVerify handshake message (
"CertificateVerify": {
  "signature algorithm": ecdsa_secp384r1_sha384
  "signature": {
    0000: 30 66 02 31 00 8F 0A 1E   3E 01 CE 4B 65 0E FA A0  0f.1....>..Ke...
    0010: 58 B3 A8 0A 25 3B 58 6D   68 5B DE E5 C6 0A F2 10  X...%;Xmh[......
    0020: 1F 33 01 E9 B8 01 57 B2   D6 03 E8 A4 02 20 CD 9B  .3....W...... ..
    0030: 52 D1 B7 45 B6 02 31 00   A9 08 67 09 D5 7D D7 5C  R..E..1...g....\
    0040: CB F7 FA BD 0A 2A A0 71   98 DD D4 D2 CF 01 4E 7A  .....*.q......Nz
    0050: 7F 06 E2 95 A4 AB 4E 18   34 B0 C9 3C 6B A7 9B 4C  ......N.4..<k..L
    0060: D9 36 56 3D 5D B5 4C 16                            .6V=].L.
  }
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:16.025 EDT|Finished.java:917|Consuming server Finished handshake message (
"Finished": {
  "verify data": {
    0000: 91 54 08 B7 9F E5 54 B3   CB 0A DF 1D 8F DB 88 49  .T....T........I
    0010: DF 3A DA 6C C7 D6 9B 35   62 F5 81 9E A2 D4 A3 D2  .:.l...5b.......
  }
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:16.025 EDT|SSLCipher.java:1836|KeyLimit read side: algorithm = AES/GCM/NoPadding:KEYUPDATE
countdown value = 137438953472
javax.net.ssl|ALL|10|main|2025-07-27 20:51:16.026 EDT|X509Authentication.java:205|X509KeyManager class: io.helidon.common.tls.TlsReloadableX509KeyManager
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:16.026 EDT|CertificateMessage.java:1103|Produced client Certificate message (
"Certificate": {
  "certificate_request_context": "",
  "certificate_list": [  
  {
    "certificate" : {
      "version"            : "v3",
      "serial number"      : "1c:4b:72:0a:84:80:ee:fd",
      "signature algorithm": "Ed25519",
      "issuer"             : "CN=Files f7c6d0qpc7:8222",
      "not before"         : "2024-07-29 01:00:00.000 EDT",
      "not  after"         : "3000-01-01 01:00:00.000 EST",
      "subject"            : "CN=Files f7c6d0qpc7:8222",
      "subject public key" : "EdDSA"}
    "extensions": {
      <no extension>
    }
  },
]
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:16.029 EDT|CertificateVerify.java:1134|Produced client CertificateVerify handshake message (
"CertificateVerify": {
  "signature algorithm": ed25519
  "signature": {
    0000: F7 6F BE 83 4D 0F 8E 69   C2 92 AD F0 2C 0A DF 56  .o..M..i....,..V
    0010: 99 EC 4A 2C 31 3A 54 A2   0D ED 1C 0D 11 70 EA 66  ..J,1:T......p.f
    0020: 10 5D 50 69 C4 58 96 23   8B C2 FA 86 76 A5 50 44  .]Pi.X.#....v.PD
    0030: A3 D4 DC 1C E0 80 A6 BC   55 C4 50 04 6B E1 BF 0D  ........U.P.k...
  }
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:16.030 EDT|Finished.java:687|Produced client Finished handshake message (
"Finished": {
  "verify data": {
    0000: 65 A7 BC 44 8A 5A 4A 1B   74 94 64 63 72 AE C2 B8  e..D.ZJ.t.dcr...
    0010: 58 CD 74 54 CB D1 90 B0   4B 15 DF 5D 5B 62 6D 91  X.tT....K..][bm.
  }
}
)
javax.net.ssl|DEBUG|10|main|2025-07-27 20:51:16.030 EDT|SSLCipher.java:1987|KeyLimit write side: algorithm = AES/GCM/NoPadding:KEYUPDATE
countdown value = 137438953472
javax.net.ssl|DEBUG|82|helidon-client-0|2025-07-27 20:51:16.060 EDT|NewSessionTicket.java:566|Consuming NewSessionTicket message (
"NewSessionTicket": {
  "ticket_lifetime"      : "86,400",
  "ticket_age_add"       : "<omitted>",
  "ticket_nonce"         : "01",
  "ticket"               : {
    0000: 2A 62 4D 85 25 3D D9 27   A4 8C 3D E2 85 1C 3C DE  *bM.%=.'..=...<.
    0010: 53 D2 7D 14 F5 03 5D D2   BD A6 99 0A 55 13 FD 1B  S.....].....U...
    0020: 5B B5 DA BF BE 1E 05 28   AD 15 82 01 05 5D CD 86  [......(.....]..
    0030: 01 DB 52 3B F9 68 D0 83   46 93 95 1C 93 8A 33 E7  ..R;.h..F.....3.
    0040: 13 FA C4 5F F1 99 68 3A   4A 2C 11 69 30 CA E9 1C  ..._..h:J,.i0...
    0050: FB 5E 2A 98 D8 FE 08 02   AB 9C 24 31 6A C8 EC A9  .^*.......$1j...
    0060: D9 CD 6E 2A 5F 5A 15 C6   13 3C D3 44 05 5F 8F 69  ..n*_Z...<.D._.i
    0070: 20 9E C8 BF 59 BF D6 FF   6F 04 DB FE B8 18 4F DB   ...Y...o.....O.
    0080: 9C 91 54 37 F0 57 89 14   CD 28 25 E4 21 B1 A4 06  ..T7.W...(%.!...
    0090: D6 65 BE D7 8F 62 88 0F   A2 B3 FB 4C E7 79 15 ED  .e...b.....L.y..
    00A0: D1 2C D4 F9 4F 30 35 D8   A4 18 16 1A 60 C0 BF D5  .,..O05.....`...
    00B0: F5 BF AD 53 DB 1E BA 0A   A7 37 CE 57 92 5C 87 96  ...S.....7.W.\..
    00C0: 45 57 42 FE 38 A8 D5 B1   14 BC F7 51 B4 82 06 8F  EWB.8......Q....
    00D0: FA 9B 35 AD ED 5B 6F 71   B5 3C FA CD 61 CC 47 80  ..5..[oq.<..a.G.
    00E0: B2 5E DE BB 54 85 E5 70   45 A6 F1 73 5A 14 19 4C  .^..T..pE..sZ..L
    00F0: 6A 07 DD 56 11 B9 6E 2A   96 0F DF 18 12 6B 6E 97  j..V..n*.....kn.
    0100: 9A F3 00 86 83 90 25 7F   95 C2 78 CC D5 D9 AA 60  ......%...x....`
    0110: FD 7A 26 D1 E3 E1 E9 71   55 83 8C 91 54 EB C8 48  .z&....qU...T..H
    0120: 87 6F D0 EF 98 1F 46 33   1B 8E C1 AB 54 E2 A9 75  .o....F3....T..u
    0130: D5 37 BB 7E D9 48 1C A5   A9 A0 C6 14 93 56 34 57  .7...H.......V4W
    0140: BD 8C FD 3C D3 16 10 63   84 8A 9B C8 B5 10 EC 9B  ...<...c........
    0150: 52 2F 5B 58 8D 7E BA BC   43 20 F8 DB 11 7E DB F9  R/[X....C ......
    0160: AC CD A7 A3 11 27 CD 32   41 94 06 2D 8E 45 D3 4C  .....'.2A..-.E.L
    0170: F8 48 AF E3 E2 AB 22 64   BE DE 84 0B 5D 5D C6 AA  .H...."d....]]..
    0180: 81 AC 00 EB 62 41 46 DF   C9 70 1E 94 9B 33 AD FC  ....bAF..p...3..
    0190: 4E 2C 6A 5D 83 B8 1F 50   A1 58 18 E6 4D 8D 8E 9F  N,j]...P.X..M...
    01A0: 56 B9 A3 26 82 CD A5 AC   5B 2E 82 39 51 A5 19 8A  V..&....[..9Q...
    01B0: D8 82 CB 8F DF 03 19 91   F7 B4 D5 EC 80 98 1A 2F  .............../
    01C0: 4E 8F 36 2B 11 00 C7 23   A9 E0 A6 12 3C A0 FD C5  N.6+...#....<...
    01D0: 24 69 5A 9D 87 DF BB 1D   D2 46 8C EE 87 27 70 D3  $iZ......F...'p.
    01E0: 21 DF 76 E5 E6 67 E5 78   30 81 79 69 99 90 64 22  !.v..g.x0.yi..d"
    01F0: CE 78 19 AE D3 14 F3 B4   2B CA 1C 4F 60 90 46 19  .x......+..O`.F.
    0200: BB 2C 7F 6B D6 95 2E 26   D0 0C F4 04 C4 B1 7F 7E  .,.k...&........
    0210: 6D 6C 7F 9D 8F CB CA 1F   4C 5A AF E5 9D 77 59 22  ml......LZ...wY"
    0220: 1E DB 06 06 E3 C1 92 DD   26 B2 3C F6 CD B1 8E 81  ........&.<.....
    0230: 56 9A 3A 7F 95 A1 5E 1B   6D EF 09 B2 81 78 1D 76  V.:...^.m....x.v
    0240: E6 2B 31 CD 52 71 A1 A8   DD FC E5 60 7C 1B E7 40  .+1.Rq.....`...@
    0250: 67 DE 7D 2D 08 AC 08 4C   15 79 2E E9 51 76 24 C6  g..-...L.y..Qv$.
    0260: DC DA 2C E3 6B DD 8B 7D   A2 C3 F0 91 8D ED CC FF  ..,.k...........
    0270: 7D 97 AB F5 FD 1D 16 74   00 2B 8B 36 99 F7 1D 0D  .......t.+.6....
    0280: 46 FF C8 66 02 5E 2E B4   85 87 ED F0 1A 45 DA FA  F..f.^.......E..
    0290: 44 73 EE 1C D3 40 E3 41   48 B9 3F 0B FA EA DE 44  Ds...@.AH.?....D
    02A0: C5 82 2B F0 B5 B7 7D C5   7B 80 E6 AD 70 7B 55 BC  ..+.........p.U.
    02B0: BA 0E 02 AD E2 E5 46 B5   A6 AF C3 D8 D1 DD 67 B7  ......F.......g.
    02C0: A1 78 80 1F E5 3D 03 A0   82 B8 3F 1F 1E 92 19 D7  .x...=....?.....
    02D0: E9 90 66 F8 A5 E0 DC 7C   0B 11 F1 99 D2 51 7C FF  ..f..........Q..
    02E0: 81 20 85 FF CB 3A C6 DD   14 6A E2 CF 00 7D 0C 80  . ...:...j......
    02F0: DC 0A E0 EC B6 1B 7F C1   BC 70 D2 38 D6 E2 F3 55  .........p.8...U
    0300: 65 B3 6F 10 87 0A 8A 0E   E2 9C CA 42 49 A6 D3 38  e.o........BI..8
    0310: 58 43 B7 00 5D 4A DE A7   17 BC AB B6 4E A9 D8 FA  XC..]J......N...
    0320: C8 87 36 2B 5F D3 12 2F   50 87 CF 17 2D 83 5F 96  ..6+_../P...-._.
    0330: AB 73 F4 85 76 5A 70 ED   20 1E CA DB C2 3C 10 82  .s..vZp. ....<..
    0340: 95 A8 46 DC 12 09 55 A2   B1 EA 52 5E D1 96 2E DD  ..F...U...R^....
    0350: AB 55 80 16 3D F6 68 39   27 E3 F9 1F C3 DA 85 B2  .U..=.h9'.......
    0360: E7 18 AE DB 01 A4 81 9C   9E B5 0C F0 10 A5 C2 3D  ...............=
    0370: 96 FE BF 7B 6A A3 B6 35   B1 E9 D7 0D 
  }  "extensions"           : [
    <no extension>
  ]
}
)
javax.net.ssl|DEBUG|82|helidon-client-0|2025-07-27 20:51:16.060 EDT|Alert.java:232|Received alert message (
"Alert": {
  "level"      : "warning",
  "description": "close_notify"
}
)
Jul 27, 2025 8:51:16 PM io.helidon.common.socket.SocketContext log
SEVERE: [client 0x604c5de8] java.net.SocketException: Broken pipe
java.io.UncheckedIOException: java.net.SocketException: Broken pipe
	at io.helidon.common.buffers.FixedBufferData.writeTo(FixedBufferData.java:74)
	at io.helidon.common.buffers.CompositeArrayBufferData.writeTo(CompositeArrayBufferData.java:48)
	at io.helidon.common.socket.PlainSocket.write(PlainSocket.java:136)
	at io.helidon.webclient.api.TcpClientConnection$BufferedDataWriter.writeNow(TcpClientConnection.java:368)
	at io.helidon.http.http2.Http2ConnectionWriter.noLockWrite(Http2ConnectionWriter.java:200)
	at io.helidon.http.http2.Http2ConnectionWriter.lockedWrite(Http2ConnectionWriter.java:173)
	at io.helidon.http.http2.Http2ConnectionWriter.splitAndWrite(Http2ConnectionWriter.java:210)
	at io.helidon.http.http2.Http2ConnectionWriter.writeData(Http2ConnectionWriter.java:65)
	at io.helidon.webclient.http2.Http2ClientStream.write(Http2ClientStream.java:507)
	at io.helidon.webclient.http2.Http2ClientStream.splitAndWrite(Http2ClientStream.java:497)
	at io.helidon.webclient.http2.Http2ClientStream.writeData(Http2ClientStream.java:343)
	at io.helidon.webclient.grpc.GrpcClientCall.lambda$startStreamingThreads$1(GrpcClientCall.java:153)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.VirtualThread.run(VirtualThread.java:329)
Caused by: java.net.SocketException: Broken pipe
	at java.base/sun.nio.ch.SocketDispatcher.write0(Native Method)
	at java.base/sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:62)
	at java.base/sun.nio.ch.NioSocketImpl.tryWrite(NioSocketImpl.java:394)
	at java.base/sun.nio.ch.NioSocketImpl.implWrite(NioSocketImpl.java:410)
	at java.base/sun.nio.ch.NioSocketImpl.write(NioSocketImpl.java:440)
	at java.base/sun.nio.ch.NioSocketImpl$2.write(NioSocketImpl.java:819)
	at java.base/java.net.Socket$SocketOutputStream.write(Socket.java:1195)
	at java.base/sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:345)
	at java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1304)
	at io.helidon.common.buffers.FixedBufferData.writeTo(FixedBufferData.java:71)
	... 14 more
```

Changing line 288 to `List.of("h2")` removes the warning and lets the client connect to a Netty gRPC server, though the protocols should probably come from the config or the `WebClient` or something.

Given that the Helidon gRPC client is able to connect to the Helidon gRPC server, but not the Netty server, it seems likely that the Helidon server is not coordinating (or at least not strictly enforcing?) ALPN. I did not explore further.

--- a/webclient/grpc/src/main/java/io/helidon/webclient/grpc/GrpcBaseClientCall.java
+++ b/webclient/grpc/src/main/java/io/helidon/webclient/grpc/GrpcBaseClientCall.java
@@ -21,7 +21,6 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.time.Duration;
-import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
@@ -55,11 +54,11 @@ import io.helidon.webclient.api.DnsAddressLookup;
 import io.helidon.webclient.api.Proxy;
 import io.helidon.webclient.api.TcpClientConnection;
 import io.helidon.webclient.api.WebClient;
+import io.helidon.webclient.http2.Http2Client;
 import io.helidon.webclient.http2.Http2ClientConnection;
 import io.helidon.webclient.http2.Http2ClientImpl;
 import io.helidon.webclient.http2.Http2StreamConfig;
 import io.helidon.webclient.http2.StreamTimeoutException;
-import io.helidon.webclient.http2.Http2Client;
 
 import io.grpc.CallOptions;
 import io.grpc.ClientCall;
@@ -83,7 +82,6 @@ abstract class GrpcBaseClientCall<ReqT, ResT> extends ClientCall<ReqT, ResT> {
     protected static final BufferData PING_FRAME = BufferData.create("PING");
     protected static final BufferData EMPTY_BUFFER_DATA = BufferData.empty();
     protected static final int DATA_PREFIX_LENGTH = 5;
-
     protected static final Tag OK_TAG = Tag.create("grpc.status", "OK");
     protected record MethodMetrics(Counter callStarted,
                                    Timer callDuration,

